pipeline/azure-pipelines.yml

```diff
@@ -0,0 +1,215 @@
+    inputs:
+      targetType: filePath
+      filePath: $(System.DefaultWorkingDirectory)\pipeline\scripts\fetch-vsix-version.ps1
+  - task: VSBuild@1
+    displayName: Set BranchName, Sha1 and BuildNumber properties from Azdo pipeline variables
+    inputs:
+      solution: build/ChangeVersion.proj
+      msbuildArgs: /p:Sha1=$(Build.SourceVersion) /p:BranchName=$(Build.SourceBranchName) /p:BuildNumber=$(Build.BuildId)  /p:BuildConfiguration=$(BuildConfiguration)
+  - task: DownloadSecureFile@1
+    name: snk
+    displayName: Download snk file
+    inputs:
+      secureFile: c638339f-2dc0-42e1-9b9c-7ee76eda15f5
+      retryCount: 5
+  - task: DownloadSecureFile@1
+    name: pfx
+    displayName: Download pfx file
+    inputs:
+      secureFile: c2ebf861-df45-46e0-9245-6ef7ae954a8b
+      retryCount: 5
+  - task: SonarCloudPrepare@1
+    displayName: Prepare analysis on SonarCloud (VS2022 only)
+    condition: and(succeeded(), eq(variables['vsTargetVersion'], '2022'))
+    inputs:
+      SonarCloud: 4d6d3285-de8e-41bb-a346-987397025edb
+      organization: sonarsource
+      projectKey: sonarlint-visualstudio
+      projectName: SonarLint for Visual Studio
+      projectVersion: $(SONAR_PROJECT_VERSION)
+      extraProperties: "# Additional properties that will be passed to the scanner, \n# Put one key=value per line, example:\n# sonar.exclusions=**/*.bin\n#sonar.cs.vstest.reportsPaths=$(Agent.TempDirectory)/**/*.trx\n#sonar.cs.vscoveragexml.reportsPaths=$(Agent.TempDirectory)/**/*.coveragexml"
+  - task: VSBuild@1
+    displayName: Build solution SonarLint.VisualStudio.Integration.sln
+    inputs:
+      solution: $(BuildParameters.solution)
+      vsVersion: 17.0
+      msbuildArgs: /p:VsTargetVersion=$(vsTargetVersion) /p:DeployExtension=false /p:SignArtifacts=$(SHOULD_SIGN) /p:AssemblyOriginatorKeyFile="$(snk.secureFilePath)" /p:pfxCertificatePath="$(pfx.secureFilePath)" /p:pfxPassword="$(pfxPassword2021-2023)" /p:pfxSha1=$(pfxSha1-2021-2023) /p:Sha1=$(Build.SourceVersion) /p:BuildNumber=$(Build.BuildId)  $(AdditionalMSBuildArgs)
+      platform: $(BuildPlatform)
+      configuration: $(BuildConfiguration)
+      maximumCpuCount: true
+      logFileVerbosity: diagnostic
+  - task: UniversalPackages@0
+    displayName: Download WhiteSource agent
+    condition: and(succeeded(), or(eq(variables['ForceWhiteSourceScan'], 'true'), eq(variables['Build.SourceBranch'], 'refs/heads/master')))
+    inputs:
+      downloadDirectory: $(System.DefaultWorkingDirectory)\Whitesource
+      feedListDownload: 399fb241-ecc7-4802-8697-dcdd01fbb832/69e11f1b-cad7-442a-b676-37722e207221
+      packageListDownload: agent
+      versionListDownload: '*'
+  - task: PowerShell@2
+    displayName: Run WhiteSource (Mend) scan
+    condition: and(succeeded(), or(eq(variables['ForceWhiteSourceScan'], 'true'), eq(variables['Build.SourceBranch'], 'refs/heads/master')))
+    inputs:
+      targetType: inline
```

Comment:
kept this one inline because it was already a one-liner. It didn't make sense to extract the script to a file.